### DISCUSSION TITLE
Fusion后续优化方向投票讨论

### DISCUSSION THREAD

**Discussion by kyokaxin** [alibaba-fusion/next#4421] [General]
🌟🤝欢迎大家积极参与投票讨论！目前已整理了9个Fusion组件库的后续方向。每位成员最多可以点击投票给3个方向⭐️，同时也可以回复补充那些没有被提到的优化方向🔍。一起来为Fusion的发展助力💪吧！💥

快来踊跃投票🗳️和提出你的想法💡，让我们的Fusion组件库变得更加出色🌈🚀！💫🎉

一起展现你的声音📣和创造力✨，让我们共同打造一个更好的Fusion体验💪🔥！参与其中，即刻改变未来！💥💯

（投票方式：点击 🔘回复 左下角的⬆️按钮。其他想要优化的点：回复你的想法。）

**Comment by kyokaxin** [comment]
1. 类型定义：全面转向typescript, 完善类型定义说明。

**Reply by eternalsky** [reply]
对于大家最关心的这个问题，我们已经在进行积极的进展，同时也非常欢迎大家参与进来，帮助我们推进。具体可以参考 Wiki 里的 https://github.com/alibaba-fusion/next/wiki/Tech-upgrade 来查看当前进度以及参与办法。

**Comment by kyokaxin** [comment]
2. 文档完善：包括组件的用法、API、配置选项和常见问题解答等信息，帮助开发者更好地理解和使用组件。

**Comment by kyokaxin** [comment]
3. 示例完善：涵盖不同的用法和场景，帮助开发者快速上手和调试。

**Reply by xmsz-stu** [reply]
示例希望可以实时编辑和预览

**Reply by eternalsky** [reply]
这个现在就可以的

**Comment by kyokaxin** [comment]
4. 修复已知错误：Table、CascaderSelect、DatePicker2、Form、Tree组件的问题。

**Comment by kyokaxin** [comment]
5. 优化渲染性能：复杂组件或需要频繁更新的组件，可以通过优化渲染流程来提高性能。

**Reply by xmsz-stu** [reply]
这个太重要了

**Comment by kyokaxin** [comment]
6. 减少内存占用：减少组件的依赖和引用，优化构建性能。

**Comment by kyokaxin** [comment]
7. 错误处理和异常捕获：对潜在的错误和异常进行处理和捕获，提供友好的错误提示和处理方式。

**Comment by kyokaxin** [comment]
8. 设计组件：根据所提的feature、市面上其他类似组件的情况进行设计、规划更多的组件，考虑组件的用途、外观、交互方式等因素，定义组件的API和配置选项，以及与现有组件库的一致性和兼容性。

**Reply by J-Yaleen** [reply]
骨架屏+1

**Comment by kyokaxin** [comment]
9. 主题定制能力：提供更多的主题变量、样式覆盖机制和动态切换主题的功能。满足不同项目的视觉需求，提升用户体验和品牌形象。

**Reply by yee94** [reply]
这里的技术选型是什么？有预选方案吗？

**Reply by eternalsky** [reply]
目前的想法还是会基于 fusion design system 的能力，以及 css variables 的方案，但会更加丰富细节和可定制性。有更好的想法也可以在下面回复提出哈

**Reply by lime2019** [reply]
可以提供不使用 sass 前提下，修改组件类名前缀的方法吗。

**Reply by eternalsky** [reply]
@lime2019 可以开个 issue 讨论一下你的需求，这里讨论不是很方便。

**Reply by stormrise** [reply]
tailwindcss

**Comment by jasonzhuang** [comment]
10. 支持Figma插件

**Comment by eternalsky** [comment]
11. hooks 化改造

**Comment by Wuriqilang** [comment]
12.升级到React18

**Reply by stormrise** [reply]
react 17/18的兼容性
#2803 
以及peerDependencies依赖解决下

**Reply by eternalsky** [reply]
peer 依赖的问题已解决。

**Comment by zxcvbnmha** [comment]
13、移动端组件的API对齐PC组件的API

**Reply by a521wangandroid** [reply]
移动端的组件文档在哪里可以找到？
